packages/components/src/DataTable/Item/DataTableRow.tsx

```diff
@@ -139,12 +153,8 @@ export const DataTableRow = styled(DataTableRowLayout)`
 
     td,
     th {
-      background: ${({ checked, isHeaderRow, theme: { colors } }) =>
-        checked && !isHeaderRow
-          ? colors.keyAccent
-          : isHeaderRow
-          ? undefined
-          : colors.ui1};
+      background: ${({ checked, isHeaderRow, onClick, theme }) =>
+        getRowHoverColor(theme, onClick !== undefined, isHeaderRow, checked)};
```

Comment:
You can tighten up the function parameters by casting everything exactly here:

```suggestion
        getRowHoverColor(theme, Boolean(onClick), Boolean(isHeaderRow), Boolean(checked))};
```